pyxem/generators/indexation_generator.py

```diff
@@ -95,29 +100,29 @@ def correlate(self,
             # Index at all real space pixels
             mask = 1
 
```

Comment:
so I would put the `select_method_from_dict` here, but leave this bit till last.

Reply:
One question regarding this. If I move select_method_from_methods_dict here, I will have to define functions fast_correlation and normalized_correlation within the scope of this file, or import them. Do you suggest I import them from indexation_utils? 

I will then pass the chosen method on to correlate_library as a function? 


` from pyxem.utils.indexation_utils import correlate_library, fast_correlation, normalized_correlation` 

`methods_dict = { "fast_correlation" : fast_correlation,
                     "normalized_correlation" : normalized_correlation
                     }`

`my_method = get_methods_from_methods_dict(method,methods_dict,help)`

Include argument function=my_method in call to correlate_library:
`matches = signal.map(correlate_library,
                             library=library,
                             n_largest=n_largest,
                             method = method, # will still have to provide this, as it will impact which variables to compute.
                             function = my_method,
                             mask=mask,
                             inplace=False,
                             **kwargs)`

Reply:
You should import them from `indexation_utils`, yes. But then pass the string to `.correlate_library`. Doing it this way means a user can get docstrings for the matching metric at the `.correlate` stage **and** we can optimize the internal code (see my review comment at line 130 of `indexation_utils`)